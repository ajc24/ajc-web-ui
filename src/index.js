/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
