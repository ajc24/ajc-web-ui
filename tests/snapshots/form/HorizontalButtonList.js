import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { HorizontalButtonList } from '../../../src';

describe('HorizontalButtonList', () => {
  const singleButtonList = [
    {
      id: 'test-single-button-id',
      onClick: jest.fn(),
      title: 'Test Default Single Grey Button',
      tooltip: 'This is a test tooltip',
      type: 'button',
    }
  ];
  const singleSubmitButtonList = [
    {
      id: 'test-single-submit-button-id',
      onClick: jest.fn(),
      title: 'Test Submit Button',
      tooltip: 'This is a test tooltip',
      type: 'submit',
    }
  ];
  const twoButtonsList = [
    {
      id: 'test-button-1-id',
      onClick: jest.fn(),
      title: 'Test Button 1',
      tooltip: 'This is a test tooltip',
      type: 'button',
    },
    {
      id: 'test-button-2-id',
      onClick: jest.fn(),
      title: 'Test Button 2',
      tooltip: 'This is a test tooltip',
      type: 'button',
    }
  ];
  describe('Default props and rendering - Single left aligned grey button', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HorizontalButtonList buttonList={singleButtonList} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Multiple center aligned red buttons', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HorizontalButtonList colour="red" alignment="center" buttonList={twoButtonsList} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Single right aligned grey submit button', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HorizontalButtonList colour="grey" alignment="right" buttonList={singleSubmitButtonList} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Left aligned multiple red buttons with label text, default upper spacing', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HorizontalButtonList colour="red" alignment="left" buttonList={twoButtonsList} labelText="Test Buttons List Label Text" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Center aligned multiple grey buttons with label text, added upper spacing', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HorizontalButtonList colour="grey" additionalUpperSpacing={true} alignment="center" buttonList={twoButtonsList} labelText="Test Buttons List Label Text" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Right aligned multiple red buttons with label text, no upper spacing', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HorizontalButtonList colour="red" additionalUpperSpacing={false} alignment="right" buttonList={twoButtonsList} labelText="Test Buttons List Label Text" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
