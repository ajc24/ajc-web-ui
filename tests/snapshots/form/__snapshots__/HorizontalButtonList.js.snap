// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalButtonList Default props and rendering - Single left aligned grey button verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-spacing-bottom-small ajc-background-transparent ajc-width-hd-inner ajc-flex-justify-left"
>
  <button
    className="ajc-button ajc-font-default ajc-button-grey"
    id="test-single-button-id"
    onClick={[MockFunction]}
    title="This is a test tooltip"
    type="button"
  >
    Test Default Single Grey Button
  </button>
</div>
`;

exports[`HorizontalButtonList Transferred props and rendering - Center aligned multiple grey buttons with label text, added upper spacing verifies the snapshot for the component 1`] = `
Array [
  <div
    className="ajc-flex-row ajc-spacing-bottom-small ajc-width-hd-inner ajc-text-black ajc-font-default ajc-flex-justify-center ajc-spacing-top-large"
  >
    <h2
      className="ajc-font-paragraph-heading"
    >
      Test Buttons List Label Text
    </h2>
  </div>,
  <div
    className="ajc-flex-row ajc-spacing-bottom-small ajc-background-transparent ajc-width-hd-inner ajc-flex-justify-center ajc-spacing-top-reduced"
  >
    <button
      className="ajc-button ajc-font-default ajc-button-grey ajc-spacing-right-small"
      id="test-button-1-id"
      onClick={[MockFunction]}
      title="This is a test tooltip"
      type="button"
    >
      Test Button 1
    </button>
    <button
      className="ajc-button ajc-font-default ajc-button-grey"
      id="test-button-2-id"
      onClick={[MockFunction]}
      title="This is a test tooltip"
      type="button"
    >
      Test Button 2
    </button>
  </div>,
]
`;

exports[`HorizontalButtonList Transferred props and rendering - Left aligned multiple red buttons with label text, default upper spacing verifies the snapshot for the component 1`] = `
Array [
  <div
    className="ajc-flex-row ajc-spacing-bottom-small ajc-width-hd-inner ajc-text-black ajc-font-default ajc-flex-justify-left"
  >
    <h2
      className="ajc-font-paragraph-heading"
    >
      Test Buttons List Label Text
    </h2>
  </div>,
  <div
    className="ajc-flex-row ajc-spacing-bottom-small ajc-background-transparent ajc-width-hd-inner ajc-flex-justify-left ajc-spacing-top-reduced"
  >
    <button
      className="ajc-button ajc-font-default ajc-button-red ajc-spacing-right-small"
      id="test-button-1-id"
      onClick={[MockFunction]}
      title="This is a test tooltip"
      type="button"
    >
      Test Button 1
    </button>
    <button
      className="ajc-button ajc-font-default ajc-button-red"
      id="test-button-2-id"
      onClick={[MockFunction]}
      title="This is a test tooltip"
      type="button"
    >
      Test Button 2
    </button>
  </div>,
]
`;

exports[`HorizontalButtonList Transferred props and rendering - Multiple center aligned red buttons verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-spacing-bottom-small ajc-background-transparent ajc-width-hd-inner ajc-flex-justify-center"
>
  <button
    className="ajc-button ajc-font-default ajc-button-red ajc-spacing-right-small"
    id="test-button-1-id"
    onClick={[MockFunction]}
    title="This is a test tooltip"
    type="button"
  >
    Test Button 1
  </button>
  <button
    className="ajc-button ajc-font-default ajc-button-red"
    id="test-button-2-id"
    onClick={[MockFunction]}
    title="This is a test tooltip"
    type="button"
  >
    Test Button 2
  </button>
</div>
`;

exports[`HorizontalButtonList Transferred props and rendering - Right aligned multiple red buttons with label text, no upper spacing verifies the snapshot for the component 1`] = `
Array [
  <div
    className="ajc-flex-row ajc-spacing-bottom-small ajc-width-hd-inner ajc-text-black ajc-font-default ajc-flex-justify-right"
  >
    <h2
      className="ajc-font-paragraph-heading"
    >
      Test Buttons List Label Text
    </h2>
  </div>,
  <div
    className="ajc-flex-row ajc-spacing-bottom-small ajc-background-transparent ajc-width-hd-inner ajc-flex-justify-right ajc-spacing-top-reduced"
  >
    <button
      className="ajc-button ajc-font-default ajc-button-red ajc-spacing-right-small"
      id="test-button-1-id"
      onClick={[MockFunction]}
      title="This is a test tooltip"
      type="button"
    >
      Test Button 1
    </button>
    <button
      className="ajc-button ajc-font-default ajc-button-red"
      id="test-button-2-id"
      onClick={[MockFunction]}
      title="This is a test tooltip"
      type="button"
    >
      Test Button 2
    </button>
  </div>,
]
`;

exports[`HorizontalButtonList Transferred props and rendering - Single right aligned grey submit button verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-spacing-bottom-small ajc-background-transparent ajc-width-hd-inner ajc-flex-justify-right"
>
  <button
    className="ajc-button ajc-font-default ajc-button-grey"
    id="test-single-submit-button-id"
    onClick={[MockFunction]}
    title="This is a test tooltip"
    type="submit"
  >
    Test Submit Button
  </button>
</div>
`;
